core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -306,6 +308,26 @@ oppia.controller('StateResponses', [
         return choiceIndices.every(function(choiceIndex) {
           return handledAnswersArray.indexOf(choiceIndex) !== -1;
         });
+      } else if (interactionId === 'ItemSelectionInput') {
+        var numChoices = $scope.getAnswerChoices().length;
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
```

Comment:
Unlike multiple choice, the item selection interaction has three types of rules, which you can find in extensions/interactions/rules.json. They are: Equals, ContainsAtLeastOneOf, and DoesNotContainAtLeastOneOf (which should really be 'omits at least one of').

For the case of single-choice, these translate to:
- "is the same as the given choice" (which must only have one item ticked)
- "is the same as the given choice" (ditto)
- "is not any of the given choices" (this rule may have any number of items ticked).

I think what you want here is an array of booleans, one for each choice, all initially false; then run through each of the rules, and, for each rule, convert the relevant choice booleans to true if they will trigger the rule. At the end, if all the booleans are true, then there's no need for a default answer group response.

(If you're wondering why the duplication arises, btw, that's a good question! We are thinking of merging the two interactions sometime down the road.)